docs/it/design-pattern.md

```diff
@@ -0,0 +1,126 @@
+- **Miglior comprensione del codice**: Essendo conosciuti e avendo un nome inequivocabile, è possibile riconoscere i design pattern e creare un linguaggio comune quando si documenta e sviluppa il software.
+- **Soluzioni provate e testate**: Nascendo dall'esperienza congiunta di più sviluppatori, e avvalorati nel corso degli anni, i design pattern difficilmente nascondono delle sorprese, e la loro adozione può garantire una soluzione _battle-tested_ e dimostrata ad un problema specifico.
+
+Naturalmente, come ogni cosa nel mondo del software, i design pattern comportano degli svantaggi, tra cui:
+
+1. Possono generare parti di codice più complicate. Una porzione di codice intelligibile e senza logiche complesse potrebbe beneficiare dal non utilizzo dei design pattern, che ne complicherebbero la struttura. 
+2. Per essere applicati alla lettera, i design pattern necessitano che chi li adotti accetti determinate assunzioni, ma è verosimile che chi sviluppa da poco tempo possa non conoscerli e trovarsi in difficoltà nella lettura e analisi del codice.
+3. Le prestazioni spesso non coincidono con l'adozione di design pattern, in quanto lo scopo di questi ultimi è di risolvere un problema, non necessariamente in maniera ottimizzata.
+
+**Come sempre se la domanda è: meglio utilizzarli o no? La risposta è, naturalmente, dipende dal contesto.**
+
+In generale dato un problema non sempre è semplice attribuirgli, se esiste, un design pattern che lo "riconosca"; a volte infatti il limite fra uno o l'altro non è sempre definito quindi si può entrare ad esempio scomponendo il problema più complesso (Divide et impera) in altri più semplici per i quali si potrebbero riconoscere dei pattern specifici.\
+A volte invece questa scomposizione non è possibile e, nel caso, si può comunque fare il refactoring scegliendo quello più simile o anche quello più conveniente.
+
+**E' importante sottolineare che non ci sono dei vincoli stringenti che ne impongono l'utilizzo;** \
+E' possibile applicarli oppure no, si possono mescolare a piacere, chiaramente una aderenza maggiore implica la facilità di riconoscerli da parte di altri programmatori che conoscono l'argomento.
+
+Generalmente dopo un refactoring, la nostra applicazione sarà più aderente ai [principi di sviluppo software SOLID](https://it.wikipedia.org/wiki/SOLID) il che implica appunto il principalmente il primo dei vantaggi elencati soprattutto il riuso e la facilità di test.\
+Come viene definito nel design pattern possiamo definire il design pattern vero e proprio che ne demarca i suoi confini ed il client che definisce tutta la parte esterna ad esso che vi interagisce.\
```

Comment:
Frase non chiara. Cosa intendevi?

Reply:
provo a spiegarlo meglio , magari puoi darmi anche qualche spunto.
la riga 40,41 indica che non è obbligatorio applicarli ma generalmente se vengono applicati la nostra applicazione aderirà meglio ai principi SOLID ad esempio al Principio di singola responsabilità e quindi una classe avrà una sola responsabilità sul cosa fa oppure al Principio aperto/chiuso e la nostra classe sarà aperta per essere estensa, ma chiusa alle modifiche proprio per modificare il comportamento del software.
La riga 43 indica possiamo suddividere la parte che ci interessa dove identifichiamo un pattern in client , cioè quella parte che interagisce con quella che ci interessa e il pattern vero e proprio che riusciamo ad identificare con le classi che lo compongono eccetto appunto il client che viene considerato ma è esterno. In generale si cerca di non modificarlo

Reply:
Questa non l'ho capita nemmeno io, neanche dopo la spiegazione ahahahah intendi che c'è una separazione tra l'applicazione del pattern e il modo con cui vi si interagisce? Se sì, credo forse che sia una finezza non necessaria, seppur vera. Nel caso la si voglia mantenere, la frase va rivista ma non saprei come..

Reply:
E' proprio una separazione logica strutturale , prendete ad esempio la spiegazione del pattern builder su Refactoring guru (https://refactoring.guru/design-patterns/builder) , andate sul capitolo structure la classe denominata client è la parte "esterna" che con le 4 righe sopra non dovrebbe subire cambiamenti. Il client è la parte che interagisce col nostro pattern. Spero di essere stato più chiaro

Reply:
Continua a sembrarmi una frase molto contorta.. Per spiegarne il significato sarebbe necessario andare troppo nel dettaglio, personalmente farei che toglierla.. @serenasensini ?